R/viz.R

```diff
@@ -3045,8 +3066,396 @@ fancy_vplot = function(this_sample,
     labs(title = paste0(this_sample), subtitle = plot_subtitle, x = "", y = "Variant Size") +
     geom_violin(trim = FALSE, scale = "width", color = NA) +
     stat_summary(fun = mean, geom = "point", shape = 20, size = 3, color = "black") +
-    scale_fill_manual("", values = c("DEL" = "#E6856F", "INS" = "#3A8799")) +
+    scale_fill_manual("", values = c("DEL" = "#53B1FC", "INS" = "#FC9C6D")) +
     scale_y_log10() +
     theme_cowplot() +
     theme(legend.position = "none")
 }
+
+
+#' Generate sample-level ideogram with SVs and gene annotations
+#'
+#' @param this_sample Sample to be plotted (currently only accepts one sample).
+#' @param gene_annotation Annotate ideogram with a single gene.
+#' @param include_sv Set to TRUE to plot SVs (dels and ins).
+#' @param sv_count Optional parameter to summarize n variants per chromosome, inlcude_sv must be set to TRUE.
+#' @param coding_only Optional. Set to TRUE to restrict to plotting only coding mutations.
+#' @param from_flatfile If set to true the function will use flat files instead of the database.
+#' @param use_augmented Boolean statement if to use augmented maf, default is FALSE.
+#'
+#' @return Nothing.
+#' @export
+#'
+#' @examples
+#' ideogram_MYC = fancy_ideogram(this_sample = "HTMCP-01-06-00422-01A-01D", gene_annotation = "MYC", include_sv = TRUE, sv_count = TRUE, coding_only = FALSE, from_flatfile = FALSE, use_augmented = FALSE)
+#' cnv_ideogram = fancy_ideogram(this_sample = "HTMCP-01-06-00422-01A-01D")
+#'
+fancy_ideogram = function(this_sample,
+                          gene_annotation,
+                          include_sv = TRUE,
+                          sv_count = TRUE,
+                          coding_only = FALSE,
+                          from_flatfile = TRUE,
+                          use_augmented_maf = TRUE){
+
+  #plot theme
+  ideogram_theme = function(){
+    theme(legend.position = "bottom", axis.title.x = element_blank(), axis.ticks.y = element_blank(), axis.title.y = element_blank(), axis.line.y = element_blank(), axis.text.y = element_blank())
+  }
+
+  #grch37 coordinates
+  grch37_end = c(249250621, 243199373, 198022430, 191154276, 180915260, 171115067, 159138663, 146364022, 141213431, 135534747, 135006516, 133851895, 115169878, 107349540, 102531392, 90354753, 81195210, 78077248, 59128983, 63025520, 48129895, 51304566)
```

Comment:
Yes, I could add `ref/lcr-modules-references/genomes/grch37/genome_fasta/main_chromosomes.bed` to the config.yml and then get the coordinates from the bed directly. This would also allow for use of different genome projections. What do you think?

Reply:
It is already added to GAMBLR :smiley:   You can check these out:
```
GAMBLR::chromosome_arms_grch37
GAMBLR::chromosome_arms_hg38
```

Reply:
Oh, nice! That's perfect, thanks.